src/wp-includes/functions.php

```diff
@@ -6092,6 +6092,10 @@ function wp_trigger_error( $function_name, $message, $error_level = E_USER_NOTIC
 		array( 'http', 'https' )
 	);
 
+	if ( E_USER_ERROR === $error_level ) {
```

Comment:
Since this is to handle a PHP 8.4 deprecation and the minimum PHP for WordPress Core hasn't reached that yet, should we add an inline comment to explain why this error level is handled differently?

Reply:
Well, if this PR is accepted, the `E_USER_ERROR` error level will be handled differently for _all_ supported PHP versions, not just for PHP 8.4, so I wonder if adding a comment wouldn't just cause more confusion.

And aside from the PHP 8.4 deprecation which create a form of urgency to handle this, this _could_ just be considered a form of code modernization (moving away from old-fashioned `trigger_error()` to more modern Exceptions) and as a prelim to keeping the code testable, as on PHPUnit 10+ we wouldn't be able to "expect" errors like this anymore anyway.